src/software/gui/full_system/widgets/ai_control.cpp

```diff
@@ -209,49 +211,83 @@ void setupChannelSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-
-    auto on_channel_changed = [parameter](int new_value) {
-        parameter->setValue(new_value);
-    };
+    setupGoalieIDComboBox(widget, override_friendly_goalie_id_parameter, friendly_goalie_id_parameter, "friendly");
+}
 
-    // QSpinBox has 2 "valueChanged" signals that each provide different info (string vs
-    // int), so we need to static_cast to specify the integer version
-    QWidget::connect(spin_box,
-                     static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
-                     on_channel_changed);
+void setupEnemyGoalieIDComboBox(
+        Ui::AutogeneratedFullSystemMainWidget *widget,
+        std::shared_ptr<Parameter<bool>> override_enemy_goalie_id_parameter,
+        std::shared_ptr<Parameter<int>> enemy_goalie_id_parameter
+        )
+{
+    setupGoalieIDComboBox(widget, override_enemy_goalie_id_parameter, enemy_goalie_id_parameter, "enemy");
 
-    setupSpinBox(spin_box, parameter);
 }
 
-void setupEnemyGoalieIDSpinBox(Ui::AutogeneratedFullSystemMainWidget *widget,
-                               std::shared_ptr<Parameter<int>> enemy_goalie_id_parameter)
+void setupGoalieIDComboBox(
+        Ui::AutogeneratedFullSystemMainWidget *widget,
+        std::shared_ptr<Parameter<bool>> override_goalie_id_parameter,
+        std::shared_ptr<Parameter<int>> goalie_id_parameter,
+        std::string friendly_or_enemy
+        )
 {
-    auto parameter = enemy_goalie_id_parameter;
-    auto spin_box  = widget->enemy_goalie_id_spin_box;
+    QList<QString> robot_ids;
+    for (int id = 0; id < MAX_ROBOT_IDS; id++)
+    {
+        QString str = QString::number(id);
+        robot_ids.insert(id, str);
+    }
 
-    spin_box->setValue(parameter->value());
-    spin_box->setMinimum(0);
-    spin_box->setMaximum(std::numeric_limits<uint8_t>::max());
+    if (friendly_or_enemy == "friendly") {
+        widget->friendly_goalie_id_combo_box->insertItem(0, "Use GameController");
+        widget->friendly_goalie_id_combo_box->insertItems(1, (robot_ids));
+    }
+    if (friendly_or_enemy == "enemy") {
+        widget->enemy_goalie_id_combo_box->insertItem(0, "Use GameController");
+        widget->enemy_goalie_id_combo_box->insertItems(1, (robot_ids));
+    }
 
-    auto on_channel_changed = [parameter](int new_value) {
-        parameter->setValue(new_value);
-    };
+    auto on_goalie_id_changed =
+            [override_goalie_id_parameter,
+                    goalie_id_parameter](const QString &text) {
 
-    // QSpinBox has 2 "valueChanged" signals that each provide different info (string vs
-    // int), so we need to static_cast to specify the integer version
-    QWidget::connect(spin_box,
-                     static_cast<void (QSpinBox::*)(int)>(&QSpinBox::valueChanged),
-                     on_channel_changed);
+                bool override = false;
+                int goalie_id;
+                for (int id = 0; id < MAX_ROBOT_IDS; id++)
```

Comment:
this is probably causing a build failure due to mismatch comparison
```suggestion
                for (unsigned int id = 0; id < MAX_ROBOT_IDS; id++)
```

Reply:
We also don't need a loop here-> we can just check for "Use GameController" at the beginning and if that's not true then use the goalie id from the combo box